exporter/redis.go

```diff
@@ -28,13 +31,22 @@ func (e *Exporter) connectToRedis() (redis.Conn, error) {
 		options = append(options, redis.DialPassword(e.options.Password))
 	}
 
+	if e.options.PasswordMap[uri] != "" {
```

Comment:
Don't pass `uri` if this is all you need it for, instead pass `password`. Move the password check into the caller (and check both, `e.options.password` and the map and then pass it to configureOptions).

You should use `e.redisAddr` for the password lookup from the map, in that way it's the exact same format as provided by the user and can match what they put in the password file. Makes it easier to keep those two consistent.

Also, change the function signature to `func (e *Exporter) configureOptions(uri string) ...`